components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/util/APIErrorResponseHandler.java

```diff
@@ -0,0 +1,209 @@
+    private static final String UNAUTHORIZED_ERROR_MSG = "Authorization failure. Authorization information was" +
+            " invalid or missing from your request.";
+    private static final String FORBIDDEN_ERROR_MSG = "Operation is not permitted. You do not have permissions to " +
+            "make this request.";
+
+
+    /**
+     * Generate the error response according to the relevant API endpoint and the HTTP status.
+     */
+    public static void handleErrorResponse(AuthenticationContext authenticationContext, Response response, int error,
+                                            Exception e) throws IOException {
+
+        if (log.isDebugEnabled() && e != null) {
+            log.debug("Authentication Error ", e);
+        }
+        StringBuilder value = new StringBuilder(16);
+        value.append("realm user=\"");
+        if (authenticationContext != null && authenticationContext.getUser() != null) {
+            value.append(authenticationContext.getUser().getUserName());
+        }
+        value.append('\"');
+        response.setHeader(AUTH_HEADER_NAME, value.toString());
+        if (response.getRequest() != null) {
+            String uri = response.getRequest().getRequestURI();
+            uri = removeTenantDetailFromURI(uri);
+            if (isRequestFromScim2Api(uri)) { // handles only scim 2.0 API error responses.
+                handleScim2ApiErrorResponse(response, error, e);
+            } else if (isRequestFromDCREndpoint(uri)) {
+                handleDCRApiErrorResponse(response, error, e);
+            } else {
+                handleErrorResponseForCommonAPIs(response, error, e);
+            }
+        } else { // if request is null, sending a common error message
+            handleErrorResponseForCommonAPIs(response, error, e);
+        }
+    }
+
+    private static String removeTenantDetailFromURI(String uri) {
+        if (uri.startsWith("/t")) {
+            String[] uriSplit = uri.split("/");
+            StringBuilder builder = new StringBuilder();
+            if (uriSplit.length > 3) {
+                for (int i = 3; i < uriSplit.length; i++) {
+                    builder.append('/');
+                    builder.append(uriSplit[i]);
+                }
+                return builder.toString();
+            }
+        }
+        return uri;
+    }
+
+
+    private static boolean isRequestFromScim2Api(String uri) {
+
+        if (uri == null) {
+            return false;
+        }
+        return uri.startsWith("/scim2");
+    }
+
+    private static boolean isRequestFromDCREndpoint(String uri) {
+        if (uri == null) {
+            return false;
+        }
+        return uri.startsWith("/api/identity/oauth2/dcr");
+    }
+
+    private static void handleScim2ApiErrorResponse(Response response, int error, Exception e) throws IOException {
+
+        response.setStatus(error);
+        response.setHeader("Content-Type","application/json;charset=UTF-8");
```

Comment:
We should have constants for Strings like "content-Type". Let's reuse them

Reply:
I couldn't find any constant inside the project files for this